docs/BestPractices.md

```diff
@@ -0,0 +1,56 @@
+#  AWS Java SDK 2.x Best Practices
+
+Here are the best practices of using AWS Java SDK 2.x.
+
+### Reuse SDK client if possible
+
+Each SDK client maintains its own HTTP connection pool so that connections can be reused by a new request 
+to cut down the time to establish a new connection. It is recommended to share a single instance of the client to 
+avoid the overhead of having too many connection pools that are not being utilized effectively. All SDK clients are thread safe.
+
+If it is not desirable to share a client instance, invoke `client.close()` to release the resources once the client is not needed.
+
+### Close input stream from client operations
+
+For streaming operations such as `S3Client#getObject`,  if you are working with `ResponseInputStream` directly, we have the following recommendations:
+
+- Use the data from the input stream as soon as possible
+- Read all data from the stream
```

Comment:
minor: can we combine the first item and this together into something like 'Read all the data from the stream as soon as possible'?